### PR TITLE
Lib Folder pagination (backend)

### PR DESCRIPTION
**Pull request by OlegZharkov**
Some code from https://github.com/galaxyproject/galaxy/pull/9905
Backend stuff probably should not be there, so I rebased it in this PR
It allows client to ask for `limit` and `offset`

example:
http://localhost:8080/api/folders/{fodler_id}/contents?limit={int}&offset={int}

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/folder_contents.py [hunk 1]
```diff
@@ -232,7 +239,19 @@ def _load_folder_contents(self, trans, folder, include_deleted):
                 #         subfolder.api_type = 'folder'
                 #         content_items.append( subfolder )
 
-        for dataset in folder.datasets:
+        datasets_size = int(len(folder.datasets))
```
Comment (bgruening) [inline lib/galaxy/webapps/galaxy/api/folder_contents.py]: len() should already return you an int()

**Review by bgruening — COMMENTED** [review]
You can add an example as seen here: https://github.com/galaxyproject/galaxy/blob/2777db0e46f1185e47c41049abdb61f99ac1e2e7/lib/galaxy/webapps/galaxy/api/datasets.py#L88

File: lib/galaxy/webapps/galaxy/api/folder_contents.py (hunk 1 above)
Comment (OlegZharkov) [reply lib/galaxy/webapps/galaxy/api/folder_contents.py]: thanks, I will check

**Comment by OlegZharkov** [comment]
> You can add an example as seen here:
> 
> https://github.com/galaxyproject/galaxy/blob/2777db0e46f1185e47c41049abdb61f99ac1e2e7/lib/galaxy/webapps/galaxy/api/datasets.py#L88

I can do that of course. But here we already have some sort of example https://github.com/galaxyproject/galaxy/pull/10104/files#diff-4a1789d4affda47540d7b94cac91340eR35 should I add an additional one?

**Review by bgruening — APPROVED** [review]

**Comment by OlegZharkov** [comment]
thanks a lot @bgruening @bgruening @mvdbeek !!!

**Pull request merged**